unblob/processing.py

```diff
@@ -1,26 +1,32 @@
 import stat
+import statistics
 from operator import attrgetter
 from pathlib import Path
 from typing import List
 
+import plotext as plt
 from structlog import get_logger
 
-from .extractor import carve_unknown_chunks, extract_valid_chunks, make_extract_dir
+from .extractor import carve_unknown_chunks, extract_valid_chunk, make_extract_dir
+from .file_utils import iterate_file
 from .finder import search_chunks_by_priority
 from .iter_utils import pairwise
 from .logging import noformat
+from .math import shannon_entropy
 from .models import UnknownChunk, ValidChunk
 
 logger = get_logger()
 
-DEFAULT_DEPTH = 10
 
-
-def process_file(
+# TODO: this function became too complex when adding entropy calculation, but
```

Comment:
We should plan a bit about how to do this refactor considering #191 as well. It may worth to create a config dataclass in this branch for the parameters having static values (extract_root, max_depth, entropy_depth, verbose)

Reply:
Yes, #191 got this refactored a bit. There, the `process_file` function is not called recursively, its purpose is only to be called:
1. From the `cli.py` - when using unblob as CLI tool
2. Directly - when using unblob as a 3rd party library

So keeping mind the 2nd use case, I think it is not a problem to have more (default) parameters, like `max_depth` or the `entropy_depth` here. But on that PR, I already got rid off the unnecessary stuff like `root` and `current_depth`.

If this gets accepted sooner, I don't mind rebasing and fixing #191 on top of this.

Reply:
I already have code for that, which was on this branch, but #191 is the exact reason I removed it from this branch so we can discuss it further.